Objects/typeobject.c

```diff
@@ -1674,6 +1674,68 @@ type_set_doc(PyTypeObject *type, PyObject *value, void *context)
     return PyDict_SetItem(dict, &_Py_ID(__doc__), value);
 }
 
+static PyObject *
+type_get_annotate(PyTypeObject *type, void *Py_UNUSED(ignored))
+{
+    if (!(type->tp_flags & Py_TPFLAGS_HEAPTYPE)) {
+        PyErr_Format(PyExc_AttributeError, "type object '%s' has no attribute '__annotate__'", type->tp_name);
+        return NULL;
+    }
+
+    PyObject *annotate;
+    PyObject *dict = lookup_tp_dict(type);
```

Comment:
Does this return a borrowed ref? :(

Reply:
Yes. This is probably technically unsafe (the SetItem calls below could call into user code if a key object overrides `__eq__`, which could then invalidate the dict. I'll use PyType_GetDict in the new code instead.

Reply:
Thinking about it more it probably isn't unsafe: we hold a strong reference to the type object, and there's no way to invalidate the dict from Python code without deleting the type. Still, it's safer to avoid the borrowed reference API.